src/utils/updateLegacyIssuers.ts

```diff
@@ -0,0 +1,177 @@
+import {
+  DomainHierarchy,
+  DomainReader,
+  DomainTransactionFactory,
+  IRoleDefinition,
+  ResolverContractType,
+} from '@energyweb/iam-contracts';
+import { providers } from 'ethers';
+import { Methods } from '@ew-did-registry/did';
+import { Signer } from 'ethers';
+import { isValidDID } from './did';
+import { ENSRegistry__factory } from '../../ethers/factories/ENSRegistry__factory';
+import { labelhash, namehash } from './ensHash';
+import { ChainId, chainConfigs } from '../config';
+
+const { JsonRpcProvider } = providers;
+
+/**
+ * @description - Checks that role issuers of all roles under `rootDomain` contains method-specific-id and adds it if missing
+ * `signer` must own `rootDomain` on `targetChain`
+ */
+export const updateLegacyIssuers = async (
+  rootDomain: string,
+  signer: Signer,
+  chainId: ChainId,
+  dryRun = true
+) => {
+  const {
+    transactionFactory,
+    chainName,
+    provider,
+    domainHierarchy,
+    domainReader,
+    ensRegistry,
+    ensResolverV2Address,
+  } = await initDomains(signer, chainId);
+
+  let updated = new Array<any>();
```

Comment:
```suggestion
  const updated = new Array<any>();
```
We can also an actual definition of the array items and not a `any`.

Reply:
Yeah, this let/const a bit confusing in js. I removed `any` though to not overcomplicate, because `updated` is only used for logging